File: data/src/main/java/org/cryptomator/data/cloud/pcloud/PCloudFolder.java
package org.cryptomator.data.cloud.pcloud;

import org.cryptomator.domain.Cloud;
import org.cryptomator.domain.CloudFolder;

class PCloudFolder implements CloudFolder, PCloudNode {

	private final PCloudFolder parent;
	private final String name;
	private final String path;

	public PCloudFolder(PCloudFolder parent, String name, String path) {
		this.parent = parent;
		this.name = name;
		this.path = path;
	}

	@Override
	public Cloud getCloud() {
		return parent.getCloud();
	}

	@Override
	public String getName() {
		return name;
	}

	@Override
	public String getPath() {
		return path;
	}

	@Override
	public PCloudFolder getParent() {
		return parent;
	}

	@Override
	public PCloudFolder withCloud(Cloud cloud) {
		return new PCloudFolder(parent.withCloud(cloud), name, path);
	}
}

File: data/src/apiKey/java/org/cryptomator/data/cloud/pcloud/PCloudFolder.kt
package org.cryptomator.data.cloud.pcloud

import org.cryptomator.domain.Cloud
import org.cryptomator.domain.CloudFolder

open class PCloudFolder(override val parent: PCloudFolder?, override val name: String, override val path: String) : CloudFolder, PCloudNode {

	override val cloud: Cloud?
		get() = parent?.cloud

	override fun withCloud(cloud: Cloud?): PCloudFolder? {
		return PCloudFolder(parent?.withCloud(cloud), name, path)
	}
}
